### PR TITLE
[ruby] Add GraphQL 2.3 test app and changed GraphQL blocking tests behaviours for Ruby

### PR DESCRIPTION
**Pull request by vpellan**
## Motivation

We will release GraphQL Threats Protection for Ruby in the next few weeks

## Changes

Added a test app that contains GraphQL-ruby and a GraphQL schema to test GraphQL threats protection.
Only Ruby should be affected by the changes, but I've deleted "block-all-resolvers" check as it was wrong (Not present in the ruleset, using graphql.server.all_resolvers address would still result in getting the block-resolvers rule triggered.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: utils/build/docker/ruby/graphql23/config/initializers/content_security_policy.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Be sure to restart your server when you modify this file.
```
Comment (pawelchcki) [inline utils/build/docker/ruby/graphql23/config/initializers/content_security_policy.rb:1]: Small nit.
If this file is not necessary for the Rails app to work, can we remove it ?

File: utils/build/docker/ruby/graphql23/config/initializers/inflections.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Be sure to restart your server when you modify this file.
```
Comment (pawelchcki) [inline utils/build/docker/ruby/graphql23/config/initializers/inflections.rb:1]: Small nit.
If this file is not necessary for the Rails app to work, can we remove it ?

File: utils/build/docker/ruby/graphql23/config/initializers/permissions_policy.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Define an application-wide HTTP permissions policy. For further
```
Comment (pawelchcki) [inline utils/build/docker/ruby/graphql23/config/initializers/permissions_policy.rb:1]: same as above
If this file is not necessary for the Rails app to work, can we remove it ?

File: utils/build/docker/ruby/graphql23/Gemfile [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+gem "turbo-rails"
+
+# Hotwire's modest JavaScript framework [https://stimulus.hotwired.dev]
+gem "stimulus-rails"
+
+# Build JSON APIs with ease [https://github.com/rails/jbuilder]
+gem "jbuilder"
+
+# Talk with Kafka for propagation tests
+gem "rdkafka"
```
Comment (pawelchcki) [inline utils/build/docker/ruby/graphql23/Gemfile:31]: lets remove any unused dependences to speed up the installation

**Review by pawelchcki — CHANGES_REQUESTED** [review]
A few small nits - to remove unnecessary dependencies and files. 
Otherwise LGTM

**Comment by vpellan** [comment]
Hello @pawelchcki I applied your suggestions. Something to note is that I used Rails 7.1 test app as a base for this one. I previously tried to remove system_test_controller as the tests are using graphql_controller but it crashed the app, which is why I left most of the files as-is. Nevertheless, thanks for your suggestions !

File: tests/appsec/test_blocking_addresses.py [hunk 5]
```diff
@@ -584,11 +589,11 @@ def test_request_block_attack(self):
                 parameters["address"] == "graphql.server.all_resolvers"
                 or parameters["address"] == "graphql.server.resolver"
             )
-            assert rule_triggered["rule"]["id"] == (
-                "block-resolvers" if parameters["address"] == "graphql.server.resolver" else "block-all-resolvers"
-            )
+            assert rule_triggered["rule"]["id"] == "block-resolvers"
             assert parameters["key_path"] == (
-                ["userByName", "name"]
+                ["getUserByName", "0", "name"]
```
Comment (hoolioh) [inline tests/appsec/test_blocking_addresses.py:594]: I'd add a comment clarifying why the key_path is different so next implementaters know about this. In order to solve this on the long run and get consistent results we might discuss about modifying the RFC to redefine the WAF's payload format. Other than that LGTM.

**Review by hoolioh — APPROVED** [review]

**Pull request merged**